### PR TITLE
EZP-27974: Default sort parameter for sub-items table in the Detail content view tab

### PR DESCRIPTION
**Pull request by mikadamczyk**
> https://jira.ez.no/browse/EZP-27974

For a specific location, the editor can define the sort order of the sub-items. 

screen shot 2017-11-14 at 11 38 15 am

### CODE REVIEW

File: src/bundle/Controller/LocationController.php [hunk 1]
```diff
@@ -395,4 +397,51 @@ public function addAction(Request $request): Response
...
+                        /** @Desc("Location '%name%' updated.") */
+                        'location.update.success',
+                        ['%name%' => $location->getContentInfo()->name],
+                        'location'
+                    )
+                );
+
+                return new RedirectResponse($this->generateUrl('_ezpublishLocation', [
+                        'locationId' => $location->getContentInfo()->mainLocationId,
+                    ]) . '#ez-tab-location-view-details');
```
Comment (adamwojs) [inline src/bundle/Controller/LocationController.php]: I think you can use `_fragment` parameter instead of string concatenation: https://symfony.com/blog/new-in-symfony-3-2-routing-improvements

File: src/bundle/Controller/LocationController.php [hunk 2]
```diff
@@ -395,4 +397,51 @@ public function addAction(Request $request): Response
...
+     * Handles update existing location.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function updateAction(Request $request): Response
+    {
+        $form = $this->formFactory->updateLocation(
+            new LocationUpdateData()
```
Comment (webhdx) [inline src/bundle/Controller/LocationController.php]: You don't need to initialize `LocationUpdateData()` here.

File: src/bundle/Controller/LocationController.php [hunk 3]
```diff
@@ -395,4 +397,51 @@ public function addAction(Request $request): Response
...
+        );
+        $form->handleRequest($request);
+
+        $location = $form->getData()->getLocation();
+
+        if ($form->isSubmitted()) {
```
Comment (webhdx) [inline src/bundle/Controller/LocationController.php]: Should be `$form->isSubmitted() && $form->isValid()`

File: src/bundle/Controller/LocationController.php (hunk 1 above)
Comment (webhdx) [reply src/bundle/Controller/LocationController.php]: Use magic parameter `_fragment` as @adamwojs said

File: src/bundle/Resources/translations/search.en.xliff [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:jms="urn:jms:translation" version="1.2">
-  <file date="2017-11-07T21:29:28Z" source-language="en" target-language="en" datatype="plaintext" original="not.available">
+  <file date="2017-11-09T13:48:46Z" source-language="en" target-language="en" datatype="plaintext" original="not.available">
```
Comment (webhdx) [inline src/bundle/Resources/translations/search.en.xliff]: Can we clean up this PR from unrelated translation files? I think during extraction you can limit domains.

File: src/bundle/Controller/ContentViewController.php [hunk 5]
```diff
@@ -92,15 +93,23 @@ private function supplyContentActionForms(ContentView $view): void
...
+
         $locationMoveType = $this->formFactory->moveLocation(
             new LocationMoveData($location)
         );
+
         $locationTrashType = $this->formFactory->trashLocation(
             new LocationTrashData($location)
         );
+
+        $locationUpdateType = $this->formFactory->updateLocation(
```
Comment (webhdx) [inline src/bundle/Controller/ContentViewController.php]: This form should be a part of `DetailsTab` as far as I understand. Doing this way you have to pass your form object to all location view tabs.

File: src/lib/Form/Type/ContentType/SortFieldChoiceType.php [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        $this->translator = $translator;
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'choices' => $this->getSortFieldChoices(),
+            'choices_as_values' => true,
+            'translation_domain' => 'ezrepoforms_content_type',
```
Comment (webhdx) [inline src/lib/Form/Type/ContentType/SortFieldChoiceType.php]: `ezrepoforms` domain name is reserved for `ezsystems/repository-forms`. You can use `forms` domain name.

File: src/lib/Form/Type/ContentType/SortFieldChoiceType.php [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+     *
+     * @param string $sortField
+     *
+     * @return string
+     *
+     * @throws InvalidArgumentException
+     */
+    private function getSortFieldLabel(string $sortField): string
+    {
+        return $this->translator->trans(/** @Ignore */'content_type.sort_field.' . $sortField, [], 'ezrepoforms_content_type');
```
Comment (webhdx) [inline src/lib/Form/Type/ContentType/SortFieldChoiceType.php]: Doesn't it make labels unextractable by `bin/console translation:extract`? In this case you should provide an extractor. You can do this in similar fashion as I did with menu builders: https://github.com/ezsystems/ezplatform-admin-ui/blob/a90408f631e0d982db48fa2b71b9639af2428246/src/lib/Menu/MainMenuBuilder.php#L145

Comment (adamwojs) [reply src/lib/Form/Type/ContentType/SortFieldChoiceType.php]: AFAIK this is reuse of translations that are already defined in repository-forms and they have custom extractor: https://github.com/ezsystems/repository-forms/blob/master/lib/Translation/SortingTranslationExtractor.php

Comment (mikadamczyk) [reply src/lib/Form/Type/ContentType/SortFieldChoiceType.php]: Yes, this is

File: src/bundle/Resources/translations/search.en.xliff (hunk 4 above)
Comment (mikadamczyk) [reply src/bundle/Resources/translations/search.en.xliff]: Yes, I will try

**Comment by lserwatka** [comment]
@mikadamczyk could you resolve conflicts here
